src/material-experimental/form-field/testing/shared.spec.ts

```diff
@@ -0,0 +1,262 @@
+import {createFakeEvent, dispatchFakeEvent, HarnessLoader} from '@angular/cdk/testing';
+import {TestbedHarnessEnvironment} from '@angular/cdk/testing/testbed';
+import {Component, Type} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {FormControl, ReactiveFormsModule, Validators} from '@angular/forms';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {MatFormFieldHarness} from './form-field-harness';
+
+/** Shared tests to run on both the original and MDC-based form-field's. */
+export function runHarnessTests(
+    modules: Type<any>[], formFieldHarness: typeof MatFormFieldHarness, inputHarness: Type<any>,
```

Comment:
shouldn't these be `typeof MatInputHarness` and `typeof MatSelectHarness`

Reply:
There is no real need to type them as we don't use any methods of those harnesses. We just need to compare the object instances. Importing the types would add an unnecessary dep.